src/charm.py

```diff
@@ -1,69 +1,159 @@
 #!/usr/bin/env python3
 # Copyright 2021 Canonical Ltd.
 # See LICENSE file for licensing details.
-
+import yaml
+import json
 import logging
+import ops.charm
 import subprocess
+from subprocess import check_call
 from urllib.request import urlopen
 
-from ops.charm import CharmBase
 from ops.main import main
-from ops.model import BlockedStatus, MaintenanceStatus, WaitingStatus
+from ops.model import BlockedStatus, MaintenanceStatus, WaitingStatus, ActiveStatus, Relation
 from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v0.systemd import service_running, service_resume
+from mongoserver import MongoDB, MONGODB_PORT
+from typing import Optional, List
 
 logger = logging.getLogger(__name__)
 
+PEER = "mongodb"
+
 
-class MongodbOperatorCharm(CharmBase):
+class MongodbOperatorCharm(ops.charm.CharmBase):
     """Charm the service."""
 
     def __init__(self, *args):
         super().__init__(*args)
+        self._port = MONGODB_PORT
+
         self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.start, self._on_start)
 
-    def _on_install(self, _):
-        """Install prerequisites for the application"""
+    def _on_install(self, _) -> None:
+        """Handle the install event (fired on startup).
+
+        Handles the startup install event -- installs updates the apt cache, installs MongoDB.
+        """
         self.unit.status = MaintenanceStatus("installing MongoDB")
         self._add_mongodb_org_repository()
         self._install_apt_packages(["mongodb-org"])
-        self.unit.status = WaitingStatus("Waiting to start MongoDB")
 
-    def _add_mongodb_org_repository(self):
+    def _on_config_changed(self, _: ops.charm.ConfigChangedEvent) -> None:
+        """Event handler for configuration changed events."""
+        # TODO
+        # - update existing mongo configurations based on user preferences
+        # - add additional configurations as according to spec doc
+        with open("/etc/mongod.conf") as mongo_config_file:
```

Comment:
```suggestion
        with open("/etc/mongod.conf", "r") as mongo_config_file:
```

I'd prefer this to be explicit, but this is no biggie.